File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    
    for yellow_width in range(yellow,0,-1) :
        if yellow % yellow_width != 0 :
            continue
    
        yellow_height = yellow // yellow_width
        
        if 2*yellow_width + 2*yellow_height + 4 == brown :
            break
            
    tmp1 = yellow_width + 2
    tmp2 = yellow_height + 2
    
    width = max(tmp1, tmp2)
    height = min(tmp1, tmp2)
    
    return [width, height]
            
            

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int w = yellow;
        int h = 1;
        
        while (true) {
            if ((w+2) * (h+2) - w*h == brown) {
                answer[0] = w+2;
                answer[1] = h+2;
                break;
            }
            
            w--;
            while (yellow % w != 0) {
                w--;
            }
            h = yellow / w;
        }
        
        return answer;
    }
}